### PR TITLE
feat(gatsby-plugin-sharp): added option to allow sharp process to continue with errors

### LINKED ISSUE #26857: Gatsby develop build fails when transforming images with sharp
## Description

`gatsby develop` fails when transforming images with `gatsby-transformer-sharp`/`gatsby-plugin-sharp` with error: 
```zsh
VipsJpeg: Invalid SOS parameters for sequential JPEG
VipsJpeg: out of order read at line 0
VipsJpeg: Invalid SOS parameters for sequential JPEG
```

Likely that the error is related to incorrectly encoded Samsung JPEGs per this [issue](https://github.com/lovell/sharp/issues/1578) in the `sharp` repository. If the `sharp` plugins for Gatsby allowed the passing of an option for `failOnError: false` (as suggested in the [issue](https://github.com/lovell/sharp/issues/1578#issuecomment-474299429)), that might be enough to resolve this error. Or perhaps instead of that specific option, allowing the passing through of an object with any of the acceptable options as outlined in the `sharp` [docs](https://sharp.pixelplumbing.com/api-constructor#parameters).

### Steps to reproduce

This bug is very specific to sourcing and then transforming images taken on certain Samsung devices so reproducing might be difficult unfortunately. [Edit: see comment below]

Here are details on my setup: 

I created a new Gatsby site with the [using-gatsby-source-wordpress-experimental starter](https://github.com/TylerBarnes/using-gatsby-source-wordpress-experimental) made by Tyler Barnes.

I set up the appropriate plugins.

I ran `gatsby develop`. 

All images were sourced correctly and stored locally. After sourcing, images were being transformed and kept failing when reaching images photographed with a Samsung Galaxy S7.

### Expected result

It was expected that all successfully sourced images would then be successfully transformed for use with Gatsby Image

### Actual result

Images from Samsung devices failed transformation and subsequently ended the development build.

### Notes on troubleshooting

When I was trying to solve this issue myself, I was hoping that either `gatsby-plugin-sharp` or `gatsby-transformer-sharp` would have the ability to pass through more options. I saw that the specific option I needed (and referenced in the `sharp` repo linked above) was not one of the ones I could pass, so I decided to directly manipulate the media that was causing an issue.

In my media library, it was a total of 5 images that needed to be re-encoded (by importing into Photoshop, then re-exported as a JPEG). This worked for my site, but it did not seem to be a scalable option. Unless this is fixed in the `sharp` plugins, I will need to do the same for my production site. 

It's a very specific issue, but I'd rather not have a site with thousands of images crash on a development build because of a single image.

Also, I did a small non-comprehensive test by editing the `gatsby-plugin-sharp/src/process-file.js` [file](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-sharp/src/process-file.js) and changing line 83 to be: `pipeline = sharp(file, { failOnError: false });`. This also removed the error for my development build allowing it continue without crashing. The image itself was inaccessible by `localFile.childImageSharp.fluid` but I'd rather have that than not even being able to work on the site with `gatsby develop`, especially since I do receive a GraphQL error in my terminal and I'm not unaware of a broken image.

### Environment

```zsh
  System:
    OS: macOS 10.15.6
    CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 10.16.0 - ~/.nvm/versions/node/v10.16.0/bin/node
    Yarn: 1.22.5 - ~/.yarn/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v10.16.0/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 85.0.4183.83
    Firefox: 79.0
    Safari: 13.1.2
  npmPackages:
    gatsby: ^2.24.7 => 2.24.53 
    gatsby-image: ^2.4.13 => 2.4.16 
    gatsby-plugin-chakra-ui: ^0.1.4 => 0.1.4 
    gatsby-plugin-netlify-cache: ^1.2.0 => 1.2.0 
    gatsby-plugin-react-svg: ^3.0.0 => 3.0.0 
    gatsby-plugin-sass: ^2.3.12 => 2.3.12 
    gatsby-plugin-sharp: ^2.6.19 => 2.6.31 
    gatsby-source-filesystem: ^2.3.7 => 2.3.27 
    gatsby-source-wordpress-experimental: ^1.3.8 => 1.3.8 
    gatsby-transformer-sharp: ^2.5.14 => 2.5.14 
    gatsby-wordpress-experimental-inline-images: ^0.0.3 => 0.0.3 
  npmGlobalPackages:
    gatsby-cli: 2.12.93
```

### PR DESCRIPTION
**Pull request by joesanchezjr**
## Description 

Added support for the passing of a plugin option that allows the sharp process to continue with errors so that sites with large media sites don't fail on `gatsby develop` or `gatsby build`. 

This feature is optional and the default is set to fail on error (which is the current behavior)

## Related Issues

Fixes #26857

### CODE REVIEW

**Review by vladar — COMMENTED** [review]
Thanks for the PR! Can we also have a note in [README](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-sharp/README.md) about the new option?

**Comment by joesanchezjr** [comment]
Yes! I'll work on adding that note today.

**Comment by vladar** [comment]
This is so good, thank you 💜 

There were some changes in plugin sharp in master recently - we've validation for the plugin options:

https://github.com/gatsbyjs/gatsby/blob/63532f7d3626197dbcbb792227f9042039875e8c/packages/gatsby-plugin-sharp/src/gatsby-node.js#L65-L82

Can you merge master branch and add this option there as well? Sorry for the churn.

**Comment by joesanchezjr** [comment]
@vladar No problem! I've merged master and added the new option to that object.

**Review by vladar — APPROVED** [review]
Looks great, thanks 👍

**Comment by vladar** [comment]
Published in `gatsby-plugin-sharp@2.6.42`

File: packages/gatsby-plugin-sharp/src/process-file.js [hunk 1]
```diff
@@ -61,7 +61,9 @@ sharp.concurrency(cpuCoreCount())
 exports.processFile = (file, transforms, options = {}) => {
   let pipeline
   try {
-    pipeline = sharp(file)
+    pipeline = !options.failOnError
+      ? sharp(file, { failOnError: false })
+      : sharp(file)
```
Comment (muescha) [inline packages/gatsby-plugin-sharp/src/process-file.js:66]:
```javascript
pipeline = sharp(file, { failOnError: options.failOnError })
```
why not?

Comment (muescha) [reply packages/gatsby-plugin-sharp/src/process-file.js:66]: or 
```javascript
pipeline = sharp(file, { failOnError: !!options.failOnError })
```

**Pull request merged**